src/redisearch_rs/result_processor/src/counter.rs

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2006-Present, Redis Ltd.
+ * All rights reserved.
+ *
+ * Licensed under your choice of the Redis Source Available License 2.0
+ * (RSALv2); or (b) the Server Side Public License v1 (SSPLv1); or (c) the
+ * GNU Affero General Public License v3 (AGPLv3).
+*/
+
+use crate::ResultProcessor;
+
+#[derive(Debug)]
+/// A processor to track the number of entries yielded by the previous processor in the chain.
+pub struct Counter {
+    count: usize,
+}
+
+impl ResultProcessor for Counter {
+    const TYPE: ffi::ResultProcessorType = ffi::ResultProcessorType_RP_COUNTER;
+
+    fn next(
+        &mut self,
+        mut cx: crate::Context,
+        res: &mut ffi::SearchResult,
+    ) -> Result<Option<()>, crate::Error> {
+        let mut upstream = cx
+            .upstream()
+            .expect("There is no processor upstream of this counter.");
+
+        while upstream.next(res)?.is_some() {
+            self.count += 1;
+
+            // Safety: This function should only be called on initialized SearchResults. Luckily,
+            // a ResultProcessor returning `RPStatus_RS_RESULT_OK` means "Result is filled with valid data"
+            // so it is safe to call this function inside this loop.
+            #[cfg(not(test))] // can we link Rust unit tests against the redis code?
```

Comment:
We can define the symbol in our test binary, usually to a mocking implementation. 
Check out https://github.com/RediSearch/RediSearch/pull/6257/files#diff-7b00489cf4d31bb840014ef87237cfe2f519b817ebd2207285fb5d8b61b47d5cR365-R383 as a reference example.

That's why, generally speaking, we prefer to have tests under the `tests` folder rather than colocated via a `mod tests`.

Reply:
The pin projection type sadly cannot be made `pub` so the test doesn't work outside its defining crate :/.